src/jit/flowgraph.cpp

```diff
@@ -17975,12 +17987,19 @@ void                Compiler::fgSetTreeSeqHelper(GenTreePtr tree)
         break;
     }
 
-    fgSetTreeSeqFinish(tree);
+    fgSetTreeSeqFinish(tree, isLIR);
 }
 
 void
-Compiler::fgSetTreeSeqFinish(GenTreePtr tree)
+Compiler::fgSetTreeSeqFinish(GenTreePtr tree, bool isLIR)
 {
+    // If we are sequencing a node that does not appear in LIR,
+    // do not add it to the list.
+    if (isLIR && ((tree->OperGet() == GT_LIST) || tree->OperGet() == GT_ARGPLACE))
+    {
```

Comment:
> Shouldn't this just be a check on GTK_NOTLIR? 

Not quite. If we're sequencing a tree with `isLIR` set, we really expect all of the nodes in the tree to be LIR (i.e. `GenTree::IsLIR()` should return true for each node). Indeed, if we insert a non-LIR node into an LIR range, `CheckLIR()` will assert. With the current design, `GT_LIST` and `GT_ARGPLACE` are still allowable in LIR trees, just not in LIR blocks.

> though it looks like you don't have GT_LIST and GT_ARGPLACE in that set for some reason.

See above.